tools/torch/data.lua

```diff
@@ -625,39 +609,50 @@ end
 -- @param idx (int): Current index within database
 function DBSource:nextBatch (batchSize, idx)
 
-    local Images
-    if self.augOpt.crop.use then
-        Images = torch.Tensor(batchSize, self.ImageChannels, self.augOpt.crop.len, self.augOpt.crop.len)
-    else
-        Images = torch.Tensor(batchSize, self.ImageChannels, self.ImageSizeY, self.ImageSizeX)
-    end
+    local images, labels
 
-    local Labels
-    if self.label_width == 1 then
-        Labels = torch.Tensor(batchSize)
-    else
-        Labels = torch.FloatTensor(batchSize, self.label_width)
-    end
-
-    local i=0
-
-    --local mean_ptr = torch.data(self.mean)
-    local image_ind = 0
+    -- this function creates a tensor that has similar
+    -- shape to that of the provided sample plus one
+    -- dimension (batch dimension)
+    local function createBatchTensor(sample, batchSize)
+        local t
+        if type(sample) == 'number' then
+            t = torch.Tensor(batchSize)
+        else
+            shape = sample:size():totable()
+            -- add 1 dimension (batchSize)
+            table.insert(shape, 1, batchSize)
+            t = torch.Tensor(torch.LongStorage(shape))
+        end
+        return t
+    end    
 
     for i=1,batchSize do
         -- get next sample
-        im, label = self:getSample(self.shuffle, idx + i - 1)
+        local image, label = self:getSample(self.shuffle, idx + i - 1)
```

Comment:
That'd be better. Or call any output `y` (seems to be fairly standard across torch users, `t` is fine too. `im` or `image` are often too specific for your generic intentions.